scripts/jenkins/update_automation

```diff
@@ -33,6 +33,25 @@ function do_git
 
     if [ -d $a_dir/.git ] ; then
         pushd $a_dir > /dev/null
+            local n=100
+            # check counter before mkdir to be sure that we did not get the lock when it is 0
+            while [[ $n -gt 0 ]] && ! mkdir .git/dir.lock ; do
+                sleep $((3+RANDOM%10))
+                n=$((n-1))
+            done
+            if [[ $n -eq 0 ]] ; then
+                echo "Error: Failed to get git dir.lock"
+                exit 39
+            fi
+            trap "rmdir .git/dir.lock" EXIT ERR SIGINT SIGTERM
+            while [[ $(find .git -name "*.lock") ]] && [[ $n -gt 0 ]] ; do
```

Comment:
ha, (semi)good point! this part will never work because of the `dir.lock` we've just created.

for the record: this part was left here to make sure external (i.e. run from outside `update_automation`) git commands would also be handled as much as possible.